phpBB/phpbb/textformatter/s9e/mention_helper.php

```diff
@@ -25,18 +32,74 @@ class mention_helper
 	*/
 	protected $group_profile_url;
 
+	/**
+	* @var array Array of users' and groups' colors for each cached ID
+	*/
+	protected $cached_colors = [];
+
 	/**
 	* Constructor
 	*
+	* @param \phpbb\db\driver\driver_interface $db
 	* @param string $root_path
 	* @param string $php_ext
 	*/
-	public function __construct($root_path, $php_ext)
+	public function __construct($db, $root_path, $php_ext)
 	{
+		$this->db = $db;
 		$this->user_profile_url = append_sid($root_path . 'memberlist.' . $php_ext, 'mode=viewprofile&u={USER_ID}', false);
 		$this->group_profile_url = append_sid($root_path . 'memberlist.' . $php_ext, 'mode=group&g={GROUP_ID}', false);
 	}
 
+	/**
+	 * Caches colors for specified user IDs and group IDs
+	 *
+	 * @param array $user_ids
+	 * @param array $group_ids
+	 */
+	protected function get_colors($user_ids, $group_ids)
+	{
+		$this->cached_colors = [];
+		$this->cached_colors['users'] = [];
+		$this->cached_colors['groups'] = [];
+
+		if (!empty($user_ids))
+		{
+			$query = $this->db->sql_build_query('SELECT', [
+				'SELECT' => 'u.user_colour, u.user_id',
+				'FROM'   => [
+					USERS_TABLE => 'u',
+				],
+				'WHERE'  => 'u.user_id <> ' . ANONYMOUS . '
+				AND ' . $this->db->sql_in_set('u.user_type', [USER_NORMAL, USER_FOUNDER]) . '
+				AND ' . $this->db->sql_in_set('u.user_id', $user_ids),
+			]);
+			$res = $this->db->sql_query($query);
```

Comment:
A good habit is to use `$result` as a variable name.

Reply:
I thought that was common for phpBB but now I realize that this habit probably came from extensions.
Changed to `$result`.